### ISSUE TITLE
Unhandled Exception on CRUD::setAccessCondition()

### ISSUE THREAD

**Issue by MarioVillani** [Laravel-Backpack/docs#590]
``` php
class CompanyCrudController extends CrudController
{
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation { destroy as traitDestroy;
    }
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;
    use \Backpack\ReviseOperation\ReviseOperation;

    protected $setupDetailsRowRoute = false;
    /**
     * Configure the CrudPanel object. Apply settings to all operations.
     * 
     * @return void
     */
    public function setup()
    {
        // Remove all permissions
        $this->crud->denyAccess(['list', 'create', 'update', 'delete', 'show', 'revise']);

        CRUD::setAccessCondition(['show', 'update'], function ($entry) {
            if(isset($entry)) {
                if (backpack_user()->hasRole('Reseller'))
                    return ($entry->created_by === backpack_user()->id) || ($entry->created_by_company === backpack_user()->id_company);
            }
            return false;
        });
```
If a reseller tries to open up a Company entity without having permissions, an exception is thrown and logged in laravel.log:
```
Accesso non autorizzato - non hai i permessi necessari per vedere questa pagina. {"userId":32,"exception":"[object] (Backpack\\CRUD\\app\\Exceptions\\AccessDeniedException(code: 0): Accesso non autorizzato - non hai i permessi necessari per vedere questa pagina. at /myapp/vendor/backpack/crud/src/app/Library/CrudPanel/Traits/Access.php:88)
```

Docs do not mention that exceptions are thrown by CRUD::setAccessCondition()(related section: https://backpackforlaravel.com/docs/6.x/crud-operations#handling-access-to-operations).
I think it would be better to show how to correctly handle that. Thank you in advance!

**Comment by karandatwani92** [comment]
Hey @pxpm 

An exception is being logged whenever the user doesn't have access. It is not only limited to `setAccessCondition`. `Crud::denyAccess` does the same.

I don't think that is intentional; even if it is, it should be a warning or info, not an error.

**Comment by pxpm** [comment]
> Hey @pxpm
> 
> An exception is being logged whenever the user doesn't have access. It is not only limited to `setAccessCondition`. `Crud::denyAccess` does the same.
> 
> I don't think that is intentional; even if it is, it should be a warning or info, not an error.

It throws a http unauthorized exception with forbidden (403) status: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403

Similar if you navigate to a non-existent page you get a 404.

If you think it's worth adding to the docs a description of what we talked here, go ahead and add a note in the docs about `denyAccess` throwing a 403 http error forbidden.

Cheers

**Comment by karandatwani92** [comment]
Hey @pxpm 
IDK if we are on the same page. I mean, this logs an error for each attempt, and we both assume it should not be saved in the log file:
![Screenshot 2024-08-03 at 4 28 23 PM](https://github.com/user-attachments/assets/47431b5b-d161-4870-8bad-ccc065880fc0)

**Comment by pxpm** [comment]
@karandatwani92 that depends on your log level settings I guess. 
In Backpack we are not doing `Log::error()` or something similar. Like I told you, you are trying to access a resource on server, say `user/1/edit` and you don't have access to that resource, we throw an http 403 exception. (abort(403)). 

Maybe you are looking to don't report some exceptions? https://laravel.com/docs/11.x/errors#ignoring-exceptions-by-type

**Comment by munjaldevelopment** [comment]
> Hey @pxpm IDK if we are on the same page. I mean, this logs an error for each attempt, and we both assume it should not be saved in the log file: ![Screenshot 2024-08-03 at 4 28 23 PM](https://private-user-images.githubusercontent.com/8214221/354812137-47431b5b-d161-4870-8bad-ccc065880fc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4NTM4ODcsIm5iZiI6MTcyMjg1MzU4NywicGF0aCI6Ii84MjE0MjIxLzM1NDgxMjEzNy00NzQzMWI1Yi1kMTYxLTQ4NzAtOGJhZC1jY2MwNjU4ODBmYzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDVUMTAyNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTBhZWQzMTkxOGVmODA4NTk0MzEyM2MxYzEwODVmNWI4MjgxN2FmOTI0YjI2NDgxYTQ4MzZhNmUzNjUxZWI5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2T_LvjpI0WviGrME9H8MnBf2H8VBD1avQZQDueWga5s)

Hi Karan, This is default functionality of "Laravel" if will save "Error" in log once any page is 403 forbidden for our reference,,

**Comment by pxpm** [comment]
> > Hey @pxpm IDK if we are on the same page. I mean, this logs an error for each attempt, and we both assume it should not be saved in the log file: ![Screenshot 2024-08-03 at 4 28 23 PM](https://private-user-images.githubusercontent.com/8214221/354812137-47431b5b-d161-4870-8bad-ccc065880fc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4NTM4ODcsIm5iZiI6MTcyMjg1MzU4NywicGF0aCI6Ii84MjE0MjIxLzM1NDgxMjEzNy00NzQzMWI1Yi1kMTYxLTQ4NzAtOGJhZC1jY2MwNjU4ODBmYzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDVUMTAyNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTBhZWQzMTkxOGVmODA4NTk0MzEyM2MxYzEwODVmNWI4MjgxN2FmOTI0YjI2NDgxYTQ4MzZhNmUzNjUxZWI5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2T_LvjpI0WviGrME9H8MnBf2H8VBD1avQZQDueWga5s)
> 
> Hi @karandatwani92, This is functionality of "Backpack" I guess. In log once any page is 403 forbidden for our reference,,
> 
> Please check this error on one of our LIVE project. ![image](https://private-user-images.githubusercontent.com/14837352/355069507-cb96ecf1-b9d4-460f-9931-0492f56a0768.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4NTQwOTYsIm5iZiI6MTcyMjg1Mzc5NiwicGF0aCI6Ii8xNDgzNzM1Mi8zNTUwNjk1MDctY2I5NmVjZjEtYjlkNC00NjBmLTk5MzEtMDQ5MmY1NmEwNzY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA1VDEwMjk1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2NjI4NGY5Y2ZjOTg2MDJiYjZlZDE0MzM4ZGFiMDZiMzUwNTZiYjJlYTc2NGQwODg0MjQ2MTAwYzRlNDA5Y2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O356pkYn2oYOct83n9OlAGNc0bB7qNTGFn4dXJUL7Es)

Hey @munjaldevelopment . We use `abort(403)`. Cheers

**Comment by pxpm** [comment]
I got back here with a fresh perspective, and I think I got what you guys where trying to say. 

I did some digging and I found that Laravel itself already excludes some status codes from being reported like 419 for eg. In that regard I think we can make our exception extend one that Laravel already don't report on, so it will not be logged like before. 

The @munjaldevelopment screenshot was very helpful, if it had some description of where I should look for it would have helped more from the start, but it is what it is. 

Thanks for the report and for sticking with us 🙏 

I will tag a new version later today that will have https://github.com/Laravel-Backpack/CRUD/pull/5642 merged. 

Cheers
